tests/build-test-report.js

```diff
@@ -254,6 +248,68 @@ function runTests(outputFilePath, verboseOutputFilePath) {
   });
 }
 
+function downloadCircleResults() {
+  console.log("downloading test results from circle ci...");
+  const VCS_TYPE = 'github';
+  const USERNAME = 'code-dot-org';
+  const PROJECT = 'JS-Interpreter';
+  const REQUEST_PATH = `https://circleci.com/api/v1.1/project/${VCS_TYPE}/${USERNAME}/${PROJECT}/${argv.circleBuild}/artifacts`;
+
+
+  return new Promise((resolve, reject) => request(
+    {
+      url: REQUEST_PATH,
+      json: true,
+    },
+    (error, response, artifacts) => {
+      if (error) {
+        throw error;
+      }
+
+      const resultFileUrls = artifacts
+        .filter(a => a.pretty_path === '$CIRCLE_ARTIFACTS/test-results-new.json')
+        .map(a => a.url);
+
+      const bar = new ProgressBar(
+        '[:bar] :current/:total',
+        {
+          curr: 0,
+          total: artifacts.length,
+        });
+
+      Promise
+        .all(
+          resultFileUrls.map(
+            url => new Promise(
+              (resolve, reject) => request(
+                {
+                  url: url,
+                  json: true,
+                },
+                (error, response, body) => {
+                  if (error) {
+                    reject(error);
+                  }
+                  bar.tick();
```

Comment:
Why shouldn't the bar advance even if there's an error fetching a file? Or after resolving the promise? With a nice `fetch` helper this could resolve to:

```javascript
Promise.all(resultFileUrls.map(url =>
  fetch(url)
    .then(response => {
      bar.tick();
      return response.body;
    })
))